### PR TITLE
See Issue 6020. Take plugin dependency into account

### PR DESCRIPTION
**Pull request by SimonUnge**
## Proposed Changes

Addresses #6020.

Comments to this issue suggested querying the node of its active plugins, which would require an RPC call. I took an approach where I include a plugins dependencies instead, and thus avoiding the need to do an RPC call. 

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #6020)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

**Review by michaelklishin — CHANGES_REQUESTED** [review]
There is a [failure](https://app.buildbuddy.io/invocation/4eccf099-313a-4d4b-bedb-fb78a181dfde?target=%2F%2Fdeps%2Frabbitmq_cli%3Arabbitmqctl_tests) in `//deps/rabbitmq_cli:rabbitmqctl_tests`:

```
  1) test enable command will also load its dependent plugins (EnablePluginsCommandTest)
     test/plugins/enable_plugins_command_test.exs:322
     Assertion with == failed
     code:  assert RabbitMQ.CLI.Core.CommandModules.load_commands(:all, %{})["add_super_stream"] ==
              RabbitMQ.CLI.Ctl.Commands.AddSuperStreamCommand
     left:  nil
     right: RabbitMQ.CLI.Ctl.Commands.AddSuperStreamCommand
     stacktrace:
       test/plugins/enable_plugins_command_test.exs:330: (test)
--max-failures reached, aborting test suite
Finished in 160.8 seconds (7.1s async, 153.7s sync)
226 tests, 1 failure
```

**Comment by michaelklishin** [comment]
To run that specific CLI suite you can use

``` sh
bazel test //deps/rabbitmq_cli:rabbitmqctl_tests --test_env TEST_FILE=test/plugins/enable_plugins_command_test.exs
```

**Comment by SimonUnge** [comment]
Hm, and that was the test-case I created. I ran the test suite with 'TEST_FILE=... make test', and made sure it worked, but I guess bazel will setup a different environment! Lesson learned, I'm looking into it.

**Comment by SimonUnge** [comment]
Added rabbitmq_stream_management to the list of plugins for the test suite which seemed to have been the issue.

**Comment by michaelklishin** [comment]
This introduces test interference with another suite that asserts on plugin enabled:

```
                  running_version: nil,
                  version: '3.12.0'
                },
                %{
                  enabled: :not_enabled,
                  name: :rabbitmq_management,
                  running: false,
                  running_version: nil,
                  version: '3.12.0'
                },
                %{
                  enabled: :not_enabled,
                  name: :rabbitmq_management_agent,
                  running: false,
                  running_version: nil,
                  version: '3.12.0'
                },
                %{
                  enabled: :enabled,
                  name: :rabbitmq_stomp,
                  running: false,
                  running_version: nil,
                  version: '3.12.0'
                },
                %{
                  enabled: :not_enabled,
                  name: :rabbitmq_stream,
                  running: false,
                  running_version: nil,
                  version: '3.12.0'
                },
                %{
                  enabled: :not_enabled,
                  name: :rabbitmq_stream_management,
                  running: false,
                  running_version: nil,
                  version: '3.12.0'
                },
                %{
                  enabled: :not_enabled,
                  name: :rabbitmq_web_dispatch,
                  running: false,
                  running_version: nil,
                  version: '3.12.0'
                }
              ],
              status: :node_down
            }
     stacktrace:
       test/plugins/list_plugins_command_test.exs:111: (test)
```

**Comment by michaelklishin** [comment]
I will look into making these test suites more robust, they inherently rely on shared state :( and need attention from time to time.

**Comment by michaelklishin** [comment]
Note that prior to version due to an issue in our Bazel Elixir toolchain, it was not possible to run CLI tests twice in a row without `bazel clean`, at least not if the first invocation had failed.

This has been addressed so working on CLI tools using Bazel is again possible.

**Comment by michaelklishin** [comment]
The plugins-related CLI tests are more robust now. The failures in https://github.com/rabbitmq/rabbitmq-server/pull/6316 *might* be related, we have to investigate.

**Comment by michaelklishin** [comment]
I can reproduce the CI failure in #6316 (that supersedes this PR) locally with both Bazel and Make:

``` shell
bazel test deps/rabbit/feature_flags_with_unpriveleged_user_SUITE
```

Investigating.

File: deps/rabbitmq_cli/lib/rabbitmq/cli/core/command_modules.ex [hunk 1]
```diff
@@ -74,8 +74,31 @@ defmodule RabbitMQ.CLI.Core.CommandModules do
           []
       end
 
+    all_plugins = PluginsHelpers.list(opts)
```
Comment (michaelklishin) [inline deps/rabbitmq_cli/lib/rabbitmq/cli/core/command_modules.ex:77]: This change means we either have to always have the plugins directory set, including in all kinds of test environments, or avoid returning an error when the directory does not exist.

**Comment by michaelklishin** [comment]
Superseded by #6316 with some extra changes to make a few (unrelated) test suites pass under Bazel.

**Pull request closed without merging**